test/basic.jl

```diff
@@ -19,6 +19,17 @@
     end
 end
 
+@testset "Supported Platforms" begin
+    all = supported_platforms()
+    opt_out_specific = supported_platforms(exclude=[Linux(:x86_64, libc=:glibc)])
+    islin(x) = typeof(x) == Linux
+    opt_out_fx = supported_platforms(exclude=islin)
+
+    @test length(all) == length(opt_out_specific)+1
+    @test !any(opt_out_specific .== [Linux(:x86_64, libc=:glibc)])
+    @test !any(opt_out_fx .== [Linux(:x86_64, libc=:glibc)])
```

Comment:
Just being paranoid here, but please add the following tests, so that it's obvious to readers from the future how `opt_out_fx` and `opt_out_specific` differ:

```
@test any(opt_out_specific .== [Linux(:i686 , libc=:glibc)])
@test !any(opt_out_fx .== [Linux(:i686 , libc=:glibc)])
```

Reply:
I actually inverted the logic on your suggestion, such that if the function returns true, it's excluded. I think it makes more sense given the keyword is `exclude`.

Should've mentioned that, apols.

Reply:
Right, but my request still stands.  `opt_out_specific` will only exclude `Linux(:x86_64, :glibc)`, but will still include other `Linux` platforms, for example, `Linux(:i686, :glibc)` whereas `opt_out_fx` will have all `Linux` platforms excluded.  I just want tests here showing that explicit difference, so that we don't break it in the future and so that someone reading this sees how the two `opt_out_*` lists differ.

Reply:
ah! I missed that you moved to `:i686`. Just updated